packages/web/src/pages/sign-up-page/pages/SignUpPage.tsx

```diff
@@ -34,30 +57,148 @@ type SignUpEmailValues = {
+  const submitHandler = useCallback(
     (values: SignUpEmailValues) => {
       const { email } = values
-      onNext({ stage: 'create-password', params: { email } })
+      checkEmailInUse(email)
+      dispatch(setValueField('email', email))
     },
-    [onNext]
+    [dispatch, checkEmailInUse]
   )
+  const formikForm = useFormik({
+    initialValues,
+    onSubmit: submitHandler,
+    validationSchema: toFormikValidationSchema(FormSchema),
+    validateOnBlur: true,
+    validateOnChange: false
+  })
+  const {
+    values,
+    handleChange,
+    handleBlur,
+    touched,
+    errors,
+    handleSubmit,
+    setErrors
+  } = formikForm
+
+  const isLoading = emailApiStatus === 'LOADING'
+
+  useEffect(() => {
+    if (
+      emailApiStatus === 'ERROR' ||
+      (emailApiStatus === 'SUCCESS' && emailExists)
+    ) {
+      // Redirect to sign in if the email exists already
+      navigate(SIGN_IN_PAGE)
+      return
+    }
+    if (emailApiStatus === 'SUCCESS') {
+      // Move onto the password page
+      onNext({ stage: 'create-password' })
+    }
+  }, [emailApiStatus, errorMessage, emailExists, onNext, navigate])
 
   return (
     <>
-      <Formik initialValues={initialValues} onSubmit={handleSubmit}>
-        <Form>
-          <h1>{messages.header}</h1>
-          <TextField name='email' label={messages.emailLabel} />
-          <Button variant={ButtonType.PRIMARY} type='submit'>
-            {messages.signUp}
-          </Button>
-        </Form>
-      </Formik>
-      <Text>{messages.haveAccount}</Text>
-      <Link to={SIGN_IN_PAGE}>{messages.signIn}</Link>
+      <form onSubmit={handleSubmit}>
+        <Flex
+          direction='column'
+          alignItems='center'
+          ph='2xl'
+          pv='4xl'
+          gap='2xl'
+        >
+          <PreloadImage
+            src={audiusLogoColored}
+            className={styles.logo}
+            alt='Audius Colored Logo'
+          />
+          <Flex
+            direction='column'
+            gap='l'
+            alignItems='flex-start'
+            w='100%'
+            className={styles.leftAlignText}
+          >
+            <Text color='heading' size='l' variant='heading' tag='h1'>
+              {messages.title}
+            </Text>
+            <Text color='default' size='l' variant='body' tag='h2'>
+              {messages.subHeader}
+            </Text>
+          </Flex>
+          <Flex direction='column' gap='l' w='100%' alignItems='flex-start'>
+            <TextInput
+              name='email'
+              autoFocus
+              autoComplete='email'
+              label={messages.emailLabel}
+              error={!!errors.email && touched.email}
+              helperText={
+                errors.email && touched.email ? errors.email : undefined
+              }
+              className={styles.w100}
+              onChange={(e) => {
+                // Clear errors on user change (Formik doesn't have an out of the box way to do this for you)
```

Comment:
I added this since Formik doesnt have anything for it by default
imo its a little nicer when the error state goes away when you're typing 🤷

Reply:
ah yeah i think we should create `TextField` component using new text-input, allows things to be much more declarative imo, should basically be a copy from existing TextField. Happy to pair on this to go over rationale.

Reply:
https://github.com/AudiusProject/audius-protocol/blob/main/packages/web/src/components/form-fields/HarmonyTextField.tsx

I added this while doing my page. Needs renaming and moving whenever we find a home for it.